### DISCUSSION TITLE
Question: How to use Playwright's page.route()?

### DISCUSSION THREAD

**Discussion by LinkPovilas** [serenity-js/serenity-js#1899] [How to... ?] [answered]
Hi, could you please tell me how to intercept/spy on network requests using Serenity/JS? I am refactoring CucumberJS + Playwright test project, but I can't figure out how to intercept a request and, for example, redirect the actor to a different URL. With Playwright I would do it like this:
``` 
  await page.route(/https:\/\/oba\.revolut\.com/, (route) =>
    route.continue({
      url: `https://some-url.com/callbacks?state=${state}`,
    }),
  );
```
I would also like to intercept a request and validate that the response body contains certain properties. Is there a way how to handle intercepting?

Thank you

https://playwright.dev/docs/api/class-route

**Comment by jan-molak** [comment, marked as answer]
You have two ways to approach it. If you're using Serenity/JS with Playwright Test, you can simply inject the `page` object into your test as it is the same object used by the default `actor`:

```typescript
import { describe, it } from '@serenity-js/playwright-test'

it('intercepts network requests', async ({ actor, page }) => {
  await page.route(/https:\/\/oba\.revolut\.com/, (route) =>
    route.continue({
      url: `https://some-url.com/callbacks?state=${state}`,
    }),
  );
  
  await actor.attemptsTo(
    // ...
  )
})
```

To access the native Playwright [`page`](https://playwright.dev/docs/api/class-page) in a Cucumber test suite, you'd need to first extract it from the [Serenity/JS `Page`](https://serenity-js.org/api/playwright/class/PlaywrightPage/) object.

You can do it like this:

```typescript
import { Page } from '@serenity-js/web'
import { PlaywrightPage } from '@serenity-js/playwright'

When('{actor} intercepts network requests', async ({ actor }) => {
  const serenityPage = await actor.answer(Page.current()) as PlaywrightPage
  const page = await serenityPage.nativePage()
  
  await page.route(/https:\/\/oba\.revolut\.com/, (route) =>
    route.continue({
      url: `https://some-url.com/callbacks?state=${state}`,
    }),
  );
})
```

Hope this helps!
